url20b0f880a5_jgraph_legacy_jgraph5_tgz-pJ8-org_jgraph_example_HelloWorldJ8/src/org/jgraph/JGraph.java

```diff
@@ -0,0 +1,3360 @@
+
+	/**
+	 * Sets antialiasing on or off based on the boolean value.
+	 * <p>
+	 * Fires a property change for the ANTIALIASED_PROPERTY.
+	 * 
+	 * @param newValue
+	 *            whether to turn antialiasing on or off
+	 */
+	public void setAntiAliased(boolean newValue) {
+		boolean oldValue = this.antiAliased;
+		this.antiAliased = newValue;
+		firePropertyChange(ANTIALIASED_PROPERTY, oldValue, newValue);
+	}
+
+	/**
+	 * Returns true if the graph is editable (if it allows cells to be edited).
+	 * 
+	 * @return true if the graph is editable
+	 */
+	public boolean isEditable() {
+		return editable;
+	}
+
+	/**
+	 * Determines whether the graph is editable. Fires a property change event
+	 * if the new setting is different from the existing setting.
+	 * <p>
+	 * Note: Editable determines whether the graph allows editing. This is not
+	 * to be confused with enabled, which allows the graph to handle mouse
+	 * events (including editing).
+	 * 
+	 * @param flag
+	 *            a boolean value, true if the graph is editable
+	 */
+	public void setEditable(boolean flag) {
+		boolean oldValue = this.editable;
+		this.editable = flag;
+		firePropertyChange(EDITABLE_PROPERTY, oldValue, flag);
+	}
+
+	/**
+	 * @return the groupsEditable
+	 */
+	public boolean isGroupsEditable() {
+		return groupsEditable;
+	}
+
+	/**
+	 * @param groupsEditable the groupsEditable to set
+	 */
+	public void setGroupsEditable(boolean groupsEditable) {
+		this.groupsEditable = groupsEditable;
+	}
+
+	/**
+	 * Returns true if the cell selection is enabled
+	 * 
+	 * @return true if the cell selection is enabled
+	 */
+	public boolean isSelectionEnabled() {
+		return selectionEnabled;
+	}
+
+	/**
+	 * Determines whether cell selection is enabled. Fires a property change
+	 * event if the new setting is different from the existing setting.
+	 * 
+	 * @param flag
+	 *            a boolean value, true if cell selection is enabled
+	 */
+	public void setSelectionEnabled(boolean flag) {
+		boolean oldValue = this.selectionEnabled;
+		this.selectionEnabled = flag;
+		firePropertyChange(SELECTIONENABLED_PROPERTY, oldValue, flag);
+	}
+
+	/**
+	 * Returns true if graph allows invalid null ports during previews
+	 * 
+	 * @return true if the graph allows invalid null ports during previews
+	 */
+	public boolean isPreviewInvalidNullPorts() {
+		return previewInvalidNullPorts;
+	}
+
+	/**
+	 * Determines whether the graph allows invalid null ports during previews
+	 * 
+	 * @param flag
+	 *            a boolean value, true if the graph allows invalid null ports
+	 *            during previews
+	 */
+	public void setPreviewInvalidNullPorts(boolean flag) {
+		this.previewInvalidNullPorts = flag;
+	}
+
+	/**
+	 * Returns the current double buffering graphics object. Checks to see if
+	 * the graph bounds has changed since the last time the off screen image was
+	 * created and if so, creates a new image.
+	 * 
+	 * @return the off screen graphics
+	 */
+	public Graphics getOffgraphics() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 2172-2196 into a separate method.